// -*- protocol-buffers -*-
//==============================================================================
/// @file demo_service.proto
/// @brief Signalling gRPC service example
/// @author Tor Slettnes <tor@slett.net>
///
/// @note When copying this file to create your own application, be sure to also
///
///     update this header information
//==============================================================================

syntax = "proto3";

package CC.Demo;

import public "demo_types.proto";
import public "signal_types.proto";
import "google/protobuf/empty.proto";

// Simple service to showcase command/response and event propagation
service Demo {
    // Send a greeting to other listeners
    rpc say_hello (Greeting) returns (google.protobuf.Empty);

    // Get the current time
    rpc get_current_time (google.protobuf.Empty) returns (TimeData);

    // Start streaing time updates at regular intervals
    rpc start_ticking (google.protobuf.Empty) returns (google.protobuf.Empty);

    // Stop streaming time updates
    rpc stop_ticking (google.protobuf.Empty) returns (google.protobuf.Empty);

    // Stream signals from the server
    rpc watch (CC.Signal.Filter) returns (stream Signal);
}
